### PR TITLE
Modernize setup.py

### PR DESCRIPTION
**Pull request by Eliulm**

### CODE REVIEW

**Comment by Eliulm** [comment]
It seems that onnx version `1.15` causes the CI to fail. Forcing to install `1.14.1` fixes the issue. Should I pin the version in the pyproject.toml?

File: setup.py [hunk 1]
```diff
@@ -1,59 +1,4 @@
 #!/usr/bin/env python3
-
-from pathlib import Path
 from setuptools import setup
 
-directory = Path(__file__).resolve().parent
-with open(directory / 'README.md', encoding='utf-8') as f:
-  long_description = f.read()
```
[the remark this answers is not in the document]
Comment (Eliulm) [reply setup.py:8]: It seems that `readme` is the equivalent for it, as declared by PEP 621.

**Comment by Eliulm** [comment]
I checked the successful CI run of #2188 and it was using a cached version of onnx 1.14.1.

**Comment by geohot** [comment]
Cool, I'm fine with pinning then, will merge when CI passes

**Comment by geohot** [comment]
Thanks!

**Pull request merged**